backend/state.py

```diff
@@ -227,15 +231,15 @@ def update_special_effect(self, special_effect, arg, param_arg_dict):
             param_arg_dict (Dictionary[Object, Object]): The arguments for the 
                 special effect.
         """
-        if not special_effect.arg:
-            new_special_effect = special_effect.apply_sfx_on_arg(arg, param_arg_dict)
-            self.special_effects.append(new_special_effect)
-            current = self.special_effects[self.special_effects.index(new_special_effect)]
-        else:
-            current = self.special_effects[self.special_effects.index(special_effect)]
+        replaced_effect = special_effect.apply_sfx_on_arg(arg, param_arg_dict)
+        completed_effect = special_effect.apply_sfx_on_arg(arg, param_arg_dict)
+        completed_effect.completed = True
+        if completed_effect in self.special_effects:
+            return
+        if not replaced_effect in self.special_effects:
+            self.special_effects.append(replaced_effect)
+        current = self.special_effects[self.special_effects.index(replaced_effect)]
         current.update(self, active=True)
```

Comment:
The reason for this is, previously, say a patty is done cooking, the special effect is removed. If we tried to cook the patty again, a new special effect will be created and will continue to update (though no effects can be visibly seen in game since the predicates have already been updated). So now, instead of removing the special effect, I set completed to true and I leave it in the state, so when the patty is cooked again, it sees that it has already been cooked and does not apply the special effect.